test/agent_writer_test.cpp

```diff
@@ -1,6 +1,7 @@
-#include "../src/writer.h"
```

Comment:
Yes, like this! It's been `git mv` 'd! And make a new writer_test.cpp that tests ExternalWriter (I know there's not much to test there, but still).